File: decompiled/sources/com/hjq/permissions/PermissionDelegateImplV14.java
package com.hjq.permissions;

import android.app.Activity;
import android.app.AppOpsManager;
import android.content.Context;
import android.content.Intent;
import android.net.VpnService;
import androidx.core.app.NotificationManagerCompat;

class PermissionDelegateImplV14 implements PermissionDelegate {
    public boolean isPermissionPermanentDenied(Activity activity, String str) {
        return false;
    }

    PermissionDelegateImplV14() {
    }

    public boolean isGrantedPermission(Context context, String str) {
        if (PermissionUtils.equalsPermission(str, Permission.NOTIFICATION_SERVICE)) {
            return isGrantedNotifyPermission(context);
        }
        if (PermissionUtils.equalsPermission(str, Permission.PACKAGE_USAGE_STATS)) {
            return isGrantedPackagePermission(context);
        }
        if (PermissionUtils.equalsPermission(str, Permission.BIND_NOTIFICATION_LISTENER_SERVICE)) {
            return isGrantedNotificationListenerPermission(context);
        }
        if (PermissionUtils.equalsPermission(str, Permission.BIND_VPN_SERVICE)) {
            return isGrantedVpnPermission(context);
        }
        if (AndroidVersion.isAndroid13() || !PermissionUtils.equalsPermission(str, Permission.POST_NOTIFICATIONS)) {
            return true;
        }
        return isGrantedNotifyPermission(context);
    }

    public Intent getPermissionIntent(Context context, String str) {
        if (PermissionUtils.equalsPermission(str, Permission.NOTIFICATION_SERVICE)) {
            return getNotifyPermissionIntent(context);
        }
        if (PermissionUtils.equalsPermission(str, Permission.PACKAGE_USAGE_STATS)) {
            return getPackagePermissionIntent(context);
        }
        if (PermissionUtils.equalsPermission(str, Permission.BIND_NOTIFICATION_LISTENER_SERVICE)) {
            return getNotificationListenerIntent(context);
        }
        if (PermissionUtils.equalsPermission(str, Permission.BIND_VPN_SERVICE)) {
            return getVpnPermissionIntent(context);
        }
        if (AndroidVersion.isAndroid13() || !PermissionUtils.equalsPermission(str, Permission.POST_NOTIFICATIONS)) {
            return PermissionUtils.getApplicationDetailsIntent(context);
        }
        return getNotifyPermissionIntent(context);
    }

    private static boolean isGrantedNotifyPermission(Context context) {
        return NotificationManagerCompat.from(context).areNotificationsEnabled();
    }

    private static Intent getNotifyPermissionIntent(Context context) {
        Intent intent;
        if (AndroidVersion.isAndroid8()) {
            intent = new Intent("android.settings.APP_NOTIFICATION_SETTINGS");
            intent.putExtra("android.provider.extra.APP_PACKAGE", context.getPackageName());
        } else {
            intent = null;
        }
        return (intent == null || !PermissionUtils.areActivityIntent(context, intent)) ? PermissionUtils.getApplicationDetailsIntent(context) : intent;
    }

    private static boolean isGrantedNotificationListenerPermission(Context context) {
        if (AndroidVersion.isAndroid4_3()) {
            return NotificationManagerCompat.getEnabledListenerPackages(context).contains(context.getPackageName());
        }
        return true;
    }

    private static Intent getNotificationListenerIntent(Context context) {
        Intent intent;
        if (AndroidVersion.isAndroid5_1()) {
            intent = new Intent("android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS");
        } else {
            intent = new Intent("android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS");
        }
        return !PermissionUtils.areActivityIntent(context, intent) ? PermissionUtils.getApplicationDetailsIntent(context) : intent;
    }

    private static boolean isGrantedPackagePermission(Context context) {
        int i;
        if (!AndroidVersion.isAndroid5()) {
            return true;
        }
        AppOpsManager appOpsManager = (AppOpsManager) context.getSystemService("appops");
        if (AndroidVersion.isAndroid10()) {
            i = appOpsManager.unsafeCheckOpNoThrow("android:get_usage_stats", context.getApplicationInfo().uid, context.getPackageName());
        } else {
            i = appOpsManager.checkOpNoThrow("android:get_usage_stats", context.getApplicationInfo().uid, context.getPackageName());
        }
        if (i == 0) {
            return true;
        }
        return false;
    }

    private static Intent getPackagePermissionIntent(Context context) {
        Intent intent;
        if (AndroidVersion.isAndroid5()) {
            intent = new Intent("android.settings.USAGE_ACCESS_SETTINGS");
            if (AndroidVersion.isAndroid10()) {
                intent.setData(PermissionUtils.getPackageNameUri(context));
            }
        } else {
            intent = null;
        }
        return (intent == null || !PermissionUtils.areActivityIntent(context, intent)) ? PermissionUtils.getApplicationDetailsIntent(context) : intent;
    }

    private static boolean isGrantedVpnPermission(Context context) {
        return VpnService.prepare(context) == null;
    }

    private static Intent getVpnPermissionIntent(Context context) {
        Intent prepare = VpnService.prepare(context);
        return (prepare == null || !PermissionUtils.areActivityIntent(context, prepare)) ? PermissionUtils.getApplicationDetailsIntent(context) : prepare;
    }
}

File: decompiled/sources/com/hjq/permissions/PermissionDelegateImplV14.kt
package com.hjq.permissions

import android.app.Activity
import android.app.AppOpsManager
import android.content.Context
import android.content.Intent
import android.net.VpnService
import androidx.core.app.NotificationManagerCompat

internal open class PermissionDelegateImplV14 : PermissionDelegate {
    override fun isPermissionPermanentDenied(activity: Activity, permission: String): Boolean = false

    override fun isGrantedPermission(context: Context, permission: String): Boolean {
        if (PermissionUtils.equalsPermission(permission, Permission.NOTIFICATION_SERVICE)) {
            return isGrantedNotifyPermission(context)
        }
        if (PermissionUtils.equalsPermission(permission, Permission.PACKAGE_USAGE_STATS)) {
            return isGrantedPackagePermission(context)
        }
        if (PermissionUtils.equalsPermission(permission, Permission.BIND_NOTIFICATION_LISTENER_SERVICE)) {
            return isGrantedNotificationListenerPermission(context)
        }
        if (PermissionUtils.equalsPermission(permission, Permission.BIND_VPN_SERVICE)) {
            return isGrantedVpnPermission(context)
        }
        if (AndroidVersion.isAndroid13() ||
            !PermissionUtils.equalsPermission(permission, Permission.POST_NOTIFICATIONS)
        ) {
            return true
        }
        return isGrantedNotifyPermission(context)
    }

    override fun getPermissionIntent(context: Context, permission: String): Intent {
        if (PermissionUtils.equalsPermission(permission, Permission.NOTIFICATION_SERVICE)) {
            return getNotifyPermissionIntent(context)
        }
        if (PermissionUtils.equalsPermission(permission, Permission.PACKAGE_USAGE_STATS)) {
            return getPackagePermissionIntent(context)
        }
        if (PermissionUtils.equalsPermission(permission, Permission.BIND_NOTIFICATION_LISTENER_SERVICE)) {
            return getNotificationListenerIntent(context)
        }
        if (PermissionUtils.equalsPermission(permission, Permission.BIND_VPN_SERVICE)) {
            return getVpnPermissionIntent(context)
        }
        if (AndroidVersion.isAndroid13() ||
            !PermissionUtils.equalsPermission(permission, Permission.POST_NOTIFICATIONS)
        ) {
            return PermissionUtils.getApplicationDetailsIntent(context)
        }
        return getNotifyPermissionIntent(context)
    }

    private fun isGrantedNotifyPermission(context: Context): Boolean {
        return NotificationManagerCompat.from(context).areNotificationsEnabled()
    }

    private fun getNotifyPermissionIntent(context: Context): Intent {
        val intent = if (AndroidVersion.isAndroid8()) {
            Intent("android.settings.APP_NOTIFICATION_SETTINGS").apply {
                putExtra("android.provider.extra.APP_PACKAGE", context.packageName)
            }
        } else {
            null
        }
        return if (intent == null || !PermissionUtils.areActivityIntent(context, intent)) {
            PermissionUtils.getApplicationDetailsIntent(context)
        } else {
            intent
        }
    }

    private fun isGrantedNotificationListenerPermission(context: Context): Boolean {
        if (AndroidVersion.isAndroid4_3()) {
            return NotificationManagerCompat
                .getEnabledListenerPackages(context)
                .contains(context.packageName)
        }
        return true
    }

    private fun getNotificationListenerIntent(context: Context): Intent {
        val intent = Intent("android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS")
        return if (!PermissionUtils.areActivityIntent(context, intent)) {
            PermissionUtils.getApplicationDetailsIntent(context)
        } else {
            intent
        }
    }

    private fun isGrantedPackagePermission(context: Context): Boolean {
        if (!AndroidVersion.isAndroid5()) {
            return true
        }
        val appOpsManager = context.getSystemService("appops") as AppOpsManager
        val checkResult = if (AndroidVersion.isAndroid10()) {
            appOpsManager.unsafeCheckOpNoThrow(
                "android:get_usage_stats",
                context.applicationInfo.uid,
                context.packageName
            )
        } else {
            appOpsManager.checkOpNoThrow(
                "android:get_usage_stats",
                context.applicationInfo.uid,
                context.packageName
            )
        }
        return checkResult == 0
    }

    private fun getPackagePermissionIntent(context: Context): Intent {
        val intent = if (AndroidVersion.isAndroid5()) {
            Intent("android.settings.USAGE_ACCESS_SETTINGS").apply {
                if (AndroidVersion.isAndroid10()) {
                    data = PermissionUtils.getPackageNameUri(context)
                }
            }
        } else {
            null
        }
        return if (intent == null || !PermissionUtils.areActivityIntent(context, intent)) {
            PermissionUtils.getApplicationDetailsIntent(context)
        } else {
            intent
        }
    }

    private fun isGrantedVpnPermission(context: Context): Boolean = VpnService.prepare(context) == null

    private fun getVpnPermissionIntent(context: Context): Intent {
        val intent = VpnService.prepare(context)
        return if (intent == null || !PermissionUtils.areActivityIntent(context, intent)) {
            PermissionUtils.getApplicationDetailsIntent(context)
        } else {
            intent
        }
    }
}
